src/result.rs

```diff
@@ -5,64 +5,51 @@ use std::{fmt, io};
 use thiserror::Error;
 
 /// Represents the position, as line and column, within a text format ion document.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum TextPosition {
     /// No text position available; implies binary format is being read.
     None,
     /// The position represented as line and column.
-    LineColumn(usize, usize),
+    LineAndColumn(usize, usize),
 }
 
-#[derive(Clone, Debug)]
+/// Position represents the location within an ion document where an error has been
+/// identified. For all formats `byte_offset` will contain the number of bytes into the document
+/// that have been processed prior to encountering the error. When working with the text format,
+/// `line_column` will be updated to contain the line and column as well.
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Position {
     pub byte_offset: usize,
     pub line_column: TextPosition,
```

Comment:
I'm inclined to provide accessors instead of `pub` fields to give us:
1. The ability to prevent users from changing field values after we hand them a `Position`.
2. The ability to rename fields or change their data type down the road.

I'd be good with `pub(crate)` for dev convenience, though. What do you think?

Reply:
Can we make this whole struct `pub(crate)` for now, not just the fields? I don't recall seeing any usages that would force it to be `pub`.

Reply:
With the `IonError` type offering accessors for byte offset, line, and column? If so, I'd be ok with that.

Reply:
Thoughts on converting IonError to a struct and moving the individual errors (Incomplete, IoError, etc) into an ErrorKind, similar to the way `std::io::Error` works?

It's a pretty big change, imo, but it would let us move things like position out of the enum variants and instead require position info for any error creation, rather than relying on the variants to include it. That would make accessors make more sense to me, since the information would be expected for each error variant.

Thoughts?